### PR TITLE
Use Results as the default tab for non-staging experiments

### PR DESCRIPTION
**Pull request by razvanpapadopol**
Fixes https://github.com/Automattic/abacus/issues/742

When landing from the experiment directory or by using a `/experiments/{experimentId}` link, the default tab (`view`) should be:
 - 'Overview' for staging experiments
 - 'Results' for all other statuses
 
*This change is also somewhat related to 604-gh-Automattic/experimentation-platform (see comments there).
 
**Note:**
Before closing the original issue, we should extract thoughts from https://github.com/Automattic/abacus/issues/742 to new issues.
  
## How has this been tested?

- Automated tests that cover added/changed functionality
- Manual testing with production data (locally)
- Manual testing with mock data (locally or on Vercel)
- Additional manual testing instructions (please specify):

### CODE REVIEW

**Review by aaronyan — APPROVED** [review]
LGTM! Works on my local machine too as expected.

**Review by aaronyan — APPROVED** [review]
LGTM! Works on my local machine too as expected.

File: src/components/experiments/single-view/ExperimentPageView.tsx [hunk 1]
```diff
@@ -139,6 +139,11 @@ export default function ExperimentPageView({
 
   const experimentIdSlug = createIdSlug(experimentId, experiment?.name || '')
 
+  // When landing from experiment directory or by using a '/experiments/{experimentId}' link, the initial view is:
+  // - 'Overview' for staging experiments
+  // - 'Results' for all other statuses
+  const view = viewFromPath || (!!experiment && (canEditInWizard ? ExperimentView.Overview : ExperimentView.Results))
```
Comment (jessie-ross) [inline src/components/experiments/single-view/ExperimentPageView.tsx:145]: Can we change this to redirect to the view if the view doesn't exist, I think that's a cleaner solution and will mean the URL will always reference the same page:
```
if (!view) {
    history.replace(`${location.pathname}/${(!!experiment && (canEditInWizard ? ExperimentView.Overview : ExperimentView.Results)}`)
}
```
Probably in a useEffect.

**Review by jessie-ross — CHANGES_REQUESTED** [review]

File: src/components/experiments/single-view/ExperimentPageView.tsx (hunk 1 above)
Comment (razvanpapadopol) [reply src/components/experiments/single-view/ExperimentPageView.tsx]: Thanks for the suggestion, @jessie-ross ! Implemented in https://github.com/Automattic/abacus/pull/766/commits/a3abee5a9c85d4852f2fcf0a557910be667cd19e together with a bit of logic cleanup and it works really well 🙌🏼

**Review by jessie-ross — APPROVED** [review]
Thanks for adding the redirect in!

**Pull request merged**